tests/resources/test_second_quantization.py

```diff
@@ -15,9 +15,116 @@
 Unit tests for functions needed for resource estimation with the double factorization method.
 """
 import pytest
+
 import pennylane as qml
 from pennylane import numpy as np
 
+one_h2 = np.array([[-1.25330961e00, 3.46833673e-13], [3.46944695e-13, -4.75069041e-01]])
+
+two_h2 = np.array(
+    [
+        [
+            [[6.74755872e-01, -4.00346423e-13], [-4.00290912e-13, 6.63711349e-01]],
+            [[-4.00207645e-13, 1.81210478e-01], [1.81210478e-01, -3.69482223e-13]],
+        ],
+        [
+            [[-4.00263156e-13, 1.81210478e-01], [1.81210478e-01, -3.69482223e-13]],
+            [[6.63711349e-01, -3.69482223e-13], [-3.69260178e-13, 6.97651447e-01]],
+        ],
+    ]
+)
+
+df = qml.resources.DoubleFactorization(one_h2, two_h2)
+
+
+@pytest.mark.parametrize(
+    ("one", "two", "error", "tol_factor", "tol_eigval", "br", "alpha", "beta"),
+    [
+        (one_h2, two_h2, 0.0016, 1.0e-5, 1.0e-5, 7, 10, 20),
+    ],
+)
+def test_df_params(one, two, error, tol_factor, tol_eigval, br, alpha, beta):
+    r"""Test that the DoubleFactorization class initiates correct parameters."""
+    est = qml.resources.DoubleFactorization(one, two)
+    assert np.allclose(est.one_electron, one)
+    assert np.allclose(est.two_electron, two)
+    assert np.allclose(est.error, error)
+    assert np.allclose(est.tol_factor, tol_factor)
+    assert np.allclose(est.tol_eigval, tol_eigval)
+    assert np.allclose(est.br, br)
+    assert np.allclose(est.alpha, alpha)
+    assert np.allclose(est.beta, beta)
+
+
+@pytest.mark.parametrize(
+    ("one", "two", "n", "factors", "eigvals", "eigvecs", "rank_r", "rank_m", "rank_max"),
+    [
+        (
+            one_h2,
+            two_h2,
+            4,
+            np.array(
```

Comment:
Where do we cross-check results from? Might be nice to mention that (if calculated manually).

Reply:
They come from openfermion, added this. This test is actually identical to what we had before for the ``factorize`` function.